### ISSUE TITLE
Connecting/disconnecting multiple times causes server to crash

### ISSUE THREAD

**Issue by anderm** [CatalystCode/3dtoolkit#173]
Connecting and disconnecting a client multiple times from a server, will eventually cause the server to crash. 

Repro steps:
Run any sample servers locally or on a remote VM 
Open any client sample (DirectX, web etc.) and connect/disconnect multiple times with the same server
After 4-10 tries, the server will freeze.

**Comment by bengreenier** [comment]
@anderm in this case, server refers to a 3dtoolkit application server, correct? (not TURN server, signaling server, etc).

**Comment by anderm** [comment]
@bengreenier that is correct, it refers to any rendering server sample (spinning cube, multithreaded etc.)
